segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -519,6 +523,46 @@ public void mergeSegments(MergeSegments mergeSegments) {
                     });
     }
 
+    @Override
+    public void mergeSegmentsBatch(WireCommands.MergeSegmentsBatch mergeSegments) {
+        final String operation = "mergeSegmentsBatch";
+        List<String> sources = mergeSegments.getSourceSegmentIds();
+        for (String s : sources) {
+            if (!verifyToken(s, mergeSegments.getRequestId(), mergeSegments.getDelegationToken(), operation)) {
```

Comment:
This doesn't seem right from a permissions point of view. Why would the permission to merge a segment be gated on the source rather than the destination?
I think the permissions should be read on the source and readModify on the destination. I think this and mergeSegments were just copying the pattern used here for the other operations in this file as they are all read-only. Take a look at AppendProcessor for how it readModity.

Reply:
Thanks for bringing this up. Will fix as recommended.